x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileService.java

```diff
@@ -470,7 +491,34 @@ void searchVersionedDocumentForSubject(Subject subject, ActionListener<Versioned
         });
     }
 
-    private void createNewProfile(Subject subject, String uid, ActionListener<Profile> listener) throws IOException {
+    private static final String INVALID_USERNAME_MESSAGE = "Security domain [%s] is configured to use literal username. "
+        + "As a result, creating new user profile requires the username to be at least 1 and no more than 490 characters. "
+        + "The username can contain alphanumeric characters (a-z, A-Z, 0-9), spaces, punctuation, "
+        + "and printable symbols in the Basic Latin (ASCII) block.";
+
+    private ValidationException validateUsername(Subject subject) {
+        final RealmDomain realmDomain = subject.getRealm().getDomain();
```

Comment:
I wonder if it would be possilble to have the two new domain settings inside the `RealmDomain`.
I think it would simplify some stuff. 
This is a suggestion for when we get to spend some time on refactorings.

Reply:
I thought about that and decided against it. It could simplify the check in ProfileService. But the benefit isn't significant. Also these information becomes something sent across the wire which feels weird because:
1. They are not needed outside of the local node.
2. There are some similarity between`RealmRef`/`RealmConfig` and `RealmDomain`/`DomainConfig` in that the Config class is a separate thing from the reference class. If we ever add more settings under domain, it helps to separate them earlier.